extensions/amp-redbull-player/0.1/amp-redbull-player.js

```diff
@@ -0,0 +1,298 @@
+import {dict} from '../../../src/utils/object';
+import {disableScrollingOnIframe} from '../../../src/iframe-helper';
+import {getData, listen} from '../../../src/event-helper';
+import {installVideoManagerForDoc} from '../../../src/service/video-manager-impl';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {once} from '../../../src/utils/function';
+import {removeElement} from '../../../src/dom';
+import {userAssert} from '../../../src/log';
+import addAttributesToElement from './utils/addAttributesToElement';
+import getPlayerParams from './utils/getPlayerParams';
+
+/** @private @const */
+const TAG = 'amp-redbull-player';
+
+/** @private @const */
+const ANALYTICS_EVENT_TYPE_PREFIX = 'video-custom-';
+
+/** @private @const */
+const SANDBOX = [
+  SandboxOptions.ALLOW_SCRIPTS,
+  SandboxOptions.ALLOW_SAME_ORIGIN,
+  SandboxOptions.ALLOW_POPUPS,
+  SandboxOptions.ALLOW_POPUPS_TO_ESCAPE_SANDBOX,
+  SandboxOptions.ALLOW_TOP_NAVIGATION_BY_USER_ACTIVATION,
+];
+
+/**
+ * @return {!RegExp}
+ * @private
+ */
+const getAnalyticsEventTypePrefixRegex = once(
+  () => new RegExp(`^${ANALYTICS_EVENT_TYPE_PREFIX}`)
+);
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpRedBullPlayer extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    const el = addAttributesToElement(element);
+    super(el);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {!UnlistenDef|null} */
+    this.unlistenFrame_ = null;
+
+    /** @private {string} */
+    this.tagId_ = null;
+
+    /** @private {string} */
+    this.locale_ = null;
+
+    /**
+     * @param {!Event} e
+     * @return {*} TODO(#23582): Specify return type
+     * @private
+     */
+    this.boundOnMessage_ = e => this.onMessage_(e);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    const {element} = this;
+    /*
+     * Required Paramaters: data-video-id, data-skin
+     * Break creation if required parameters are missing!
+     */
+    userAssert(
+      element.getAttribute('data-videoid'),
+      'The data-videoid attribute is required for %s',
+      element
+    );
+
+    userAssert(
+      element.getAttribute('data-skin'),
+      'The data-skin attribute is required for %s',
+      element
+    );
+    this.tagId_ = element.getAttribute('id') || '';
+    this.locale_ = element.getAttribute('data-locale') || 'global';
+
+    installVideoManagerForDoc(element);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const {element} = this;
+    const origin = 'https://player.redbull.com/amp/amp-iframe.html?';
+    // const origin = 'http://localhost:9000/amp-iframe.html?';
+    const params = `${getPlayerParams(element)}&ampTagId=${
+      this.tagId_
+    }&locale=${this.locale_}`;
+    const src = `${origin}${params}`;
+
+    const name = JSON.stringify(this.getMetadata_());
```

Comment:
Is the metadata consumed by the page inside the frame? If not, then just set an empty or generic name and remove the `getMetadata_` method.